File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Consegna Legrottaglie Gionata/Kalk Java/Pentagono.java
import java.util.Vector;
public class Pentagono extends PoligonoConvesso {
	protected double lati[]={0,0,0,0,0};
	protected int nLati=5;
	
	private void set_lati(Vector<Vertice> v) {
		  int x=0,y=1;
		  while(x<v.size()){
		      if(y==v.size())y=0;
		      lati[x] = Math.sqrt((Math.pow((v.get(y).get_X()-v.get(x).get_X()),2)+Math.pow(v.get(y).get_Y()-v.get(x).get_Y(),2)));
		      x++; y++;
		    }
	}
	public Pentagono(Vector<Vertice> V,boolean b) {
		super(V,5,"Pentagono",b);
		this.set_lati(V);
	}

	public Pentagono(Vector<Vertice> V) {
		super(V,5,"Pentagono");
	}

	Pentagono(Pentagono T){
		super(T);
		this.set_lati(T.Lista);
		}

	@Override
	public String toString() {
		  String s ="Pentagono: ";

		  for (Vertice i : this.Lista) {
	            s+= i.toString();
		  }
		  return s;
	}

	@Override
	public String get_nomeClasse() {
		return "Pentagono";
	}

	@Override
	public double get_perimetro() {
		double p=0;
		for(double n: this.lati) {
			p+=n;
		}
		return p;
	}

	@Override
	public double get_area() {
		  int a=0,b=1, c=2;
		  double area=0;
		  while(c < this.Lista.size()){
			  Vector<Vertice> temp= new Vector<Vertice>();
			  temp.add(Lista.get(a));
			  temp.add(Lista.get(b));
			  temp.add(Lista.get(c));
			  Triangolo t=new Triangolo(temp);
		      area+= t.get_area();
		      c++;b++;
		    }
		  return area;
	}
	@Override
	public double get_lato(int i) {
		return this.lati[i];
	}
	@Override
	public int get_nLati() {
		return this.nLati;
	}
	@Override
	public Pentagono clone() {
		return new Pentagono(this);
	}

}

File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Consegna Legrottaglie Gionata/Kalk c++/calcolatrice/PoligoniConvessi/pentagono.cpp
#include "pentagono.h"
#include "triangolo.h"
Pentagono::Pentagono(const Pentagono & T):PoligonoConvesso(T){set_lati(T.Lista);}


Pentagono::Pentagono(const QVector<Vertice>&V,bool b):PoligonoConvesso(V,5,"Pentagono",b){
      set_lati(V);
}

double Pentagono::get_perimetro()const{
  return lati[0]+lati[1]+lati[2]+lati[3]+lati[4];
}

double Pentagono::get_area()const{
  int a=0,b=1, c=2;
  double area=0;
  while(c < Lista.size()){
      area+= Triangolo({Lista[a], Lista[b], Lista[c]}).get_area();
      c++;b++;
    }
  return area;
}

ostream& operator<< (ostream& os,const Pentagono& T){
 os<<T.toString().toUtf8().constData();
 return os;
}

void Pentagono::set_lati(const QVector<Vertice>& v){
  int x=0,y=1;
  while(x<v.size()){
      if(y==v.size())y=0;
      lati[x] = std::sqrt((std::pow((v[y].get_X()-v[x].get_X()),2)+std::pow(v[y].get_Y()-v[x].get_Y(),2)));
      x++; y++;
    }
}

QString Pentagono::toString() const{
  QString s ="Pentagono: ";
  for(auto it=Lista.begin();it!=Lista.end();it++){
    s+= it->toString();
    }

  return s;
}

QString Pentagono::get_nomeClasse()const{
    return "Pentagono";
}
double  Pentagono::get_lati(unsigned int i)const{
  return lati[i];
}
unsigned int Pentagono::get_nLati()const{
  return nLati;
}
